src/test/lib/operators/operator_performance_data_test.cpp

```diff
@@ -202,6 +202,48 @@ TEST_F(OperatorPerformanceDataTest, JoinHashStepRuntimes) {
   }
 }
 
+TEST_F(OperatorPerformanceDataTest, JoinHashBloomFilterReductions) {
+  const auto table_a = load_table("resources/test_data/tbl/int_int2.tbl", 2);
+  const auto table_wrapper_a = std::make_shared<TableWrapper>(table_a);
+  table_wrapper_a->never_clear_output();
+  table_wrapper_a->execute();
+
+  const auto table_b = load_table("resources/test_data/tbl/int_int_shuffled.tbl", 2);  // larger than int_int.tbl
+  const auto table_wrapper_b = std::make_shared<TableWrapper>(table_b);
+  table_wrapper_b->never_clear_output();
+  table_wrapper_b->execute();
+
+  // Inner join case: checking that the number of stored positions (required for all join types except semi/anti) is
+  // larger than zero. Depending on the Bloom filter implementation (which is currently depending on the
+  // implementation of std::hash), we might store values and positions in the hash map that do not have a matching
+  // partner. Thus, we only check for the existence of the minimum number of values and positions.
+  const auto inner_join = std::make_shared<JoinHash>(
+      table_wrapper_a, table_wrapper_b, JoinMode::Inner,
+      OperatorJoinPredicate{ColumnIDPair(ColumnID{0}, ColumnID{0}), PredicateCondition::Equals});
+  inner_join->execute();
+
+  const auto& inner_perf = dynamic_cast<JoinHash::PerformanceData&>(*inner_join->performance_data);
+  EXPECT_GE(inner_perf.probe_side_materialized_values, 3ul);  // values 2,6,2
+  EXPECT_GE(inner_perf.build_side_hash_table_size, 2ul);      // values 2,6
+  EXPECT_GE(inner_perf.build_side_position_count, 3ul);       // positions 1,2,3
```

Comment:
Wouldn't this test pass because of `_GE` if we either did no bloom filtering or the bloom filtering was not reported?

Reply:
Yes to "no bloom filtering". But I don't get your second point.

Reply:
Pretty much the same thing. If, instead of reporting the "bloom-filtered" values, we just reported the input values, this test would also pass.

Reply:
That's right. Would you prefer to simply check for values being > 0 (ensure they have been set **somehow**)?

Since we have no handle on the Bloom filtering as long as we use `std::` stuff everywhere, an exact check shouldn't be possible (at least if we assume there is at least one value being filtered out).

Reply:
It's really safe to assume that std::hash of an integer is idempotent. It doesn't say so anywhere in the standard, but afaik that's what every implementation does. I wouldn't let that keep us from using an `EQ` test. Add a comment and we're good.

Reply:
Done.